### DISCUSSION TITLE
Maximum number of output results

### DISCUSSION THREAD

**Discussion by xujinlong2000** [NCAR/hrldas#145] [Q&A] [answered]
Hi，I want to know if there is a maximum output count for a single simulation result of a single point? Because when I run a simulation from 2015 to 2100, the prompt shows that it has been run to 2100 and can generate a restart file for 2100. However, the result file only has 24855 counts. This is my namelist.

 START_YEAR  = 2015
 START_MONTH = 01
 START_DAY   = 01
 START_HOUR  = 00
 START_MIN   = 00

 RESTART_FILENAME_REQUESTED = "/home/out/35.375_81.125/ACCESS-ESM1-5/historical/RESTART.2015010100_DOMAIN1"

 KDAY = 31410
 SPINUP_LOOPS = 0

 FORCING_NAME_T = "T2D"
 FORCING_NAME_Q = "Q2D"
 FORCING_NAME_U = "U2D"
 FORCING_NAME_V = "V2D"
 FORCING_NAME_P = "PSFC"
 FORCING_NAME_LW = "LWDOWN"
 FORCING_NAME_SW = "SWDOWN"
 FORCING_NAME_PR = "RAINRATE"

 DYNAMIC_VEG_OPTION                = 4
 CANOPY_STOMATAL_RESISTANCE_OPTION = 1
 BTR_OPTION                        = 1
 SURFACE_RUNOFF_OPTION             = 3
 SUBSURFACE_RUNOFF_OPTION          = 3
 DVIC_INFILTRATION_OPTION          = 1
 SURFACE_DRAG_OPTION               = 1
 FROZEN_SOIL_OPTION                = 1
 SUPERCOOLED_WATER_OPTION          = 1
 RADIATIVE_TRANSFER_OPTION         = 3
 SNOW_ALBEDO_OPTION                = 2
 PCP_PARTITION_OPTION              = 1
 SNOW_THERMAL_CONDUCTIVITY         = 1
 TBOT_OPTION                       = 1
 TEMP_TIME_SCHEME_OPTION           = 1
 GLACIER_OPTION                    = 1
 SURFACE_RESISTANCE_OPTION         = 1
 SOIL_DATA_OPTION                  = 1
 PEDOTRANSFER_OPTION               = 1
 CROP_OPTION                       = 1
 IRRIGATION_OPTION                 = 0
 IRRIGATION_METHOD                 = 0
 TILE_DRAINAGE_OPTION              = 0

 FORCING_TIMESTEP = 86400
 NOAH_TIMESTEP    = 10800
 OUTPUT_TIMESTEP  = 86400
 SPLIT_OUTPUT_COUNT = 99999
 SKIP_FIRST_OUTPUT = .false.
 RESTART_FREQUENCY_HOURS = 569760

 ! XSTART = 20
 ! XEND   = 20
 ! YSTART = 203
 ! YEND   = 203

 NSOIL=4
 soil_thick_input(1) = 0.10
 soil_thick_input(2) = 0.30
 soil_thick_input(3) = 0.60
 soil_thick_input(4) = 1.00

 ZLVL = 10.0

 SF_URBAN_PHYSICS = 0
 USE_WUDAPT_LCZ = 0

**Comment by xujinlong2000** [comment]
If the results are complete, there should be 31410 counts.

**Comment by cenlinhe** [comment, marked as answer]
No, there is no limit for the time counts in restart file. What is your current restart file size?  You may want to check if there is limitation in the NETCDF file setup on your computer or in your computer system. Typically, I would not suggest putting all the restart data into one single restart file. You may want to separate them into a few smaller restart files.

**Reply by xujinlong2000** [reply]
Thank you for your answer. I've split output file, but the total count of the output file(sum of each output file) is still only 24855, which does not include all the time(31410).

**Reply by SmithSimmon** [reply]
Hi, @cenlinhe . I am trying to run a simulation from 1981 to 1999 and set the SPINUP_LOOPS = 3. The output file contains a total of 12 values once a year. The restart file is also output once a year. I have provided the settings below. However, after three spinups, the model ran until in January 1992 without producing any output; only the restart file was generated.

START_YEAR = 1981
START_MONTH = 01
START_DAY = 01
START_HOUR = 00
START_MIN = 00

RESTART_FILENAME_REQUESTED = "/media/wink/output/RESTART.1981010100_DOMAIN1"

KDAY = 6938
SPINUP_LOOPS = 3

.....

FORCING_TIMESTEP = 3600
NOAH_TIMESTEP = 1800
OUTPUT_TIMESTEP = 262800

SPLIT_OUTPUT_COUNT = 12
SKIP_FIRST_OUTPUT = .false.
RESTART_FREQUENCY_HOURS = 8760

![微信图片_20240416164557](https://github.com/NCAR/hrldas/assets/162659864/fd19c5c7-a000-4477-889b-a2c91f9204ac)

**Reply by cenlinhe** [reply]
@SmithSimmon I think the issue is that you put 12 hours results in each single output file (SPLIT_OUTPUT_COUNT = 12), while you set OUTPUT_TIMESTEP = 262800 seconds, which equals to 73 hours and is not an integer multipler of 12 hours. Try setting  OUTPUT_TIMESTEP to 259200 seconds (i.e., 72 hours).

**Reply by SmithSimmon** [reply]
I am really sorry for my carelessness！I wrote the number wrong for OUTPUT_TIMESTEP. It should be 2628000,which equals 30day and 10hours. 12 timesteps put in a single output file equals 365days. So I can get a the value of a year for each output file.
